api/client/samples/anomaly_detection/sample_anomaly_detection.ipynb

```diff
@@ -0,0 +1,793 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "updated frequency: 7-day average\n",
+      "most recent transformed value: 31.418800014381635\n"
+     ]
+    }
+   ],
+   "source": [
+    "\n",
+    "def transform_by_data_type(df, current_value, frequency):\n",
+    "    '''\n",
+    "    change frequency for certain weather items\n",
+    "    '''\n",
+    "    metric_id = df.loc[0,'metric_id']\n",
+    "    if(metric_id == 2540047 or metric_id == 15531082): # 2540047 is Land temp, 15531082 is soil moisture (MEAN)\n",
+    "        df = df.fillna(df.mean())\n",
+    "        df['value'] = df['value'].rolling(7).mean()\n",
+    "        frequency = \"7-day average\"\n",
+    "    elif( metric_id == 2100031):\n",
+    "        df = df.fillna(0)\n",
+    "        df['value'] = df['value'].rolling(7).sum() # 2100031 is rainfall (SUM)\n",
+    "        frequency = \"7-day cumulative\"\n",
+    "    else:\n",
+    "        return df, current_value, frequency\n",
+    "    df = df.iloc[::7, :]\n",
+    "    df.index = pd.RangeIndex(len(df.index))\n",
+    "    current_value = df.iloc[-1]['value']\n",
+    "    return df, current_value, frequency\n",
+    "\n",
+    "\n",
+    "# Get percent change from average\n",
+    "df = pd.DataFrame(first_location_data)\n",
+    "\n",
+    "df, current_value, frequency = transform_by_data_type(df, most_recent_value, frequency)\n",
+    "\n",
+    "print(\"updated frequency:\", frequency)\n",
+    "print(\"most recent transformed value:\", current_value)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create a significance / anomaly test\n",
+    "\n",
+    "Now we need to find the lower and upper bounds for our testing for significance, we have chosen to consider two standard deviations from the mean. After that we will use the bounds to test for significant deviations from the mean and record highs and lows. This test could be replaced with any other test for significance.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "lower bound: 16.16235442286431\n",
+      "upper bound: 35.79264647397142\n"
+     ]
+    }
+   ],
+   "source": [
+    "\n",
+    "def find_lower_and_upper(df,standard_devs):\n",
+    "    '''\n",
+    "    Standard Deviation Calculations for Alerts:\n",
+    "    Using standard deviations we cand find out if recent numbers\n",
+    "    reported to the user are abnormal or to be expected\n",
+    "    Returns upper and lower bounds for a series and a given number of standard deviations\n",
+    "    '''\n",
+    "    mean = df.mean()\n",
```

Comment:
@nadiachil  thinking for NDVI we should be calculating the mean differently. Rather than the mean of the whole series, should be the mean of the n-th day of the year. And n is a multiple of 8 here. We did something with Alassane here: https://grointelligence.atlassian.net/browse/CLEWS-15693